hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -168,22 +165,44 @@ private void startNio() {
 
     // public for testing
     public AltoSocketConfig getClientSocketConfig() {
-        if (nodeEngine.getConfig().getAdvancedNetworkConfig().isEnabled()) {
-            ServerSocketEndpointConfig endpointConfig = (ServerSocketEndpointConfig) nodeEngine
-                    .getConfig()
+        validateSocketConfig();
+
+        if (config.getAdvancedNetworkConfig().isEnabled()) {
+            ServerSocketEndpointConfig endpointConfig = (ServerSocketEndpointConfig) config
                     .getAdvancedNetworkConfig()
                     .getEndpointConfigs()
                     .get(EndpointQualifier.CLIENT);
 
-            if (endpointConfig == null) {
-                return null;
-            }
-
             return endpointConfig.getAltoSocketConfig();
         }
 
         // unified socket
-        return nodeEngine.getConfig().getNetworkConfig().getAltoSocketConfig();
+        return config.getNetworkConfig().getAltoSocketConfig();
+    }
+
+    private void validateSocketConfig() {
+        AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
+        if (advancedNetworkConfig.isEnabled()) {
+            AltoSocketConfig defaultAltoSocketConfig = new AltoSocketConfig();
+            Map<EndpointQualifier, EndpointConfig> endpointConfigs = advancedNetworkConfig.getEndpointConfigs();
+
+            endpointConfigs.forEach(((endpointQualifier, endpointConfig) -> {
+                if (endpointQualifier != EndpointQualifier.CLIENT
+                        && !endpointConfig.getAltoSocketConfig().equals(defaultAltoSocketConfig)) {
+                    throw new InvalidConfigurationException(
+                            "Alto socket configuration is only available for clients ports for now.");
+                }
+            }));
+
+            if (endpointConfigs.get(EndpointQualifier.CLIENT) == null) {
+                // Advanced network is enabled yet there is no configured server socket
+                // for clients. This means cluster will run but no client ports will be
+                // created, so no clients can connect to the cluster.
+                throw new InvalidConfigurationException("Missing client server socket configuration. "
+                        + "If you have enabled alto and advanced networking, "
```

Comment:
A minor thing, but should we capitalize `alto`?

Reply:
Yes since it is a name.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/aee4b6c67d1d67f10b744cfb4aaf58ad744c31a1